### ISSUE TITLE
count.index interpolation can cause diff mismatch

### ISSUE THREAD

**Issue by cbarbour** [hashicorp/terraform#8062]
### Terraform Version
- Terraform v0.7.0 (Built from tag using go get, etc.)
- go version go1.6.3 darwin/amd64
### Affected Resource(s)
- aws_instance
- aws_iam_role
- aws_iam_instance_profile
### Terraform Configuration Files

``` hcl
variable "node_count" {
  description = "The number of EC2 instances to create. Scale this up and down to expose issues."
  type = "string"
}

provider "aws" {
  region = "us-west-2"
}

resource "aws_instance" "node" {
  count = "${var.node_count}"
  ami = "ami-7172b611" #Amazon Linux AMI 2016.03.3 (HVM), SSD Volume Type
  instance_type = "t2.micro"
}

resource "aws_iam_role" "example" {
  count = "${var.node_count}"
  name = "${aws_instance.node.*.id[count.index]}"
#  lifecycle = {
#    ignore_changes = [ "name" ]
#  }
  assume_role_policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Service": "ec2.amazonaws.com"
      },
      "Action": "sts:AssumeRole"
    }
  ]
}
EOF
}

resource "aws_iam_instance_profile" "example" {
  count = "${var.node_count}"
  name = "${aws_instance.node.*.id[count.index]}"
  roles = ["${aws_iam_role.example.*.name[count.index]}"]
#  lifecycle = {
#    ignore_changes = [ "name" ]
#  }
}
```
### Debug Output

https://gist.github.com/cbarbour/233a40be3c5b522ef38e39c9024896d2
https://gist.github.com/cbarbour/df818ce5413e6abd8a4356ec8efaa1bb
### Expected Behavior

Resources should have been created with described attributes
### Actual Behavior
1. Terraform fails with the error: `diffs didn't match during apply. This is a bug with Terraform and should be reported as a GitHub Issue.`
2. Succeeds upon retry.
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Increase `var.node_count` by 1.
### Important Factoids

The attached code is a simplified example of my problem. I also attempted to recreate the issue using Route53 records and was not successful. This seems to be related to issues I've been seeing with count interpolation of resource properties.
### References
- GH-8061
- GH-8005

**Comment by mitchellh** [comment]
This was fixed in 0.7.8. I ran your exact repro (which is fantastic by the way) and the error no longer occurs.

**Comment by cbarbour** [comment]
Thanks Mitchell. I'm really happy about all the bugs you and your folks have squashed in the past couple of weeks.
